### DISCUSSION TITLE
Cannot upgrade to 4.2.0 from 4.1.10

### DISCUSSION THREAD

**Discussion by Wenlong-Y** [mastodon/mastodon#29299] [Q&A] [answered]
When I run
 RAILS_ENV=production bundle exec rails assets:precompile
I got 
Browserslist: caniuse-lite is outdated. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme

Then I ran 

  npx update-browserslist-db@latest

then I run 
 RAILS_ENV=production bundle exec rails assets:precompile
again
I got this error

Compiling...
Compilation failed:
`isModuleDeclaration` has been deprecated, please migrate to `isImportOrExportDeclaration`
    at isModuleDeclaration (/home/mastodon/live/node_modules/babel-plugin-lodash/node_modules/@babel/types/lib/validators/generated/index.js:2740:35)
    at PluginPass.Program (/home/mastodon/live/node_modules/babel-plugin-lodash/lib/index.js:102:44)

Then I don't know what to do. 
Please help. Thanks!

**Comment by renchap** [comment]
Please update to the latest 4.2.7 version directly, do not try to install 4.2.0. You can use the 4.2.0 upgrade steps, but with the 4.2.7 codebase.

**Reply by Wenlong-Y** [reply]
Thanks for the information. Obviously, I have tried to install 4.2.0 and failed. So I checkout 4.1.10 and run bundle install and yarn install and everything seems fine. 

After seeing this, I tried 4.2.7 and using the steps in 4.2.0. First I need to change ruby version from 3.2.3 to 3.2.2  Then everything seems ok. I ran RAILS_ENV=production bundle exec rails assets:precompile.
got the message asking me to install npx and I did that; then I run it again, I got a compilation failed without any reason. Then I just ran this a few more times. It shows everything is up-to-date. 

I also got an error when running
RAILS_ENV=production bin/tootctl search deploy --reset-chewy
I assume this is not important, so I continued.

Then I restarted mastodon-web and sidekiq. But the website won't work and show the elephant error screen.

I then checked out 4.1.15 and did bundle install and yarn install. The server is back again.

Any thoughts? Thanks!

**Reply by renchap** [reply]
The errors you are getting are probably important and the reason why this is happening.

Once you checked out 4.2.7, you need to run `bundle install` and `yarn install` again.

Then you can run the precompile command, then the first db migration command. Do any of those show errors?

**Reply by Wenlong-Y** [reply]
I ran bundle install and yarn install after checking out 4.2.7 and then I ran precompile command and got
"Browserslist: caniuse-lite is outdated. Please run:
npx update-browserslist-db@latest
Why you should do it regularly: https://github.com/browserslist/update-db#readme"

So I ran npx update-browserslist-db@latest

I ran precompile again
I got compile failed. There is no reason.
I ran the database migration command, it seems ok.

**Reply by Wenlong-Y** [reply, marked as answer]
It turns out the ElasticSearch is not working. I tried everything online only to find that 2GB memory is not ok. I upgraded to 4GB and ElasticSearch works now. The upgrade to 4.2.7 works too following the instruction of 4.2.0. Everything works. 

Somehow ElasticSearch stopped when I was upgrading. Before the last step I have to restart ElasticSearch. After all that I just downgraded to 2GB. Everything still works. Mastodon is 4.2.7 now.
